files/image_config/coredump/coredump-config.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+DISABLE_COREDUMP_CONF="/etc/sysctl.d/50-disable-coredump.conf"
+
+if [ "$(redis-cli -n 4 HGET "COREDUMP|config" "enabled")" = "false" ] ; then
+   echo "kernel.core_pattern=" > ${DISABLE_COREDUMP_CONF}
```

Comment:
What does this mean ?

Can you please explain the impact of disable?

Reply:
When the coredump admin mode is disabled in config db, core files will not be generated. We are creating a sysctl entry to disable core dump.

Reply:
Can you give me a use case, where you would not want core file dump ?

Core file dump implies that some unexpected error occurred or user explicitly creating one with kill for a reason. In either case, the dump is *required* for analysis.

If this is the only purpose of coredump-config.service, I don't see a need.

Reply:
1. corefiles can be a space hog
2. Multiple corefiles may be generated if a process is in infinite loop
3. corefiles may contain sensitive information, so some applications may not want it to be recorded. 
4. We plan to re-use coredump-config.service for enable/disable of kernel coredump as well. Also there may be additional parameters that you would like to configure w.r.t core files (e.g limit on the size of core file). Current mode of operation is we chose some fixed numbers. But future extensions may make them configurable. To start with, we are providing a framework to enable/disable the feature.

Reply:
There is already another initiative to do core file rotation and limit the count of core files at any time per process. Turning off is definitely not the solution.

The limitation on count / size is also from Broadcom only. I need to look for it.

Reply:
In fact you are referring that PR #468, which has the following.

"	 a.  Support per-process core file rotation and archiving to optimize disk space "

Reply:
I am referring to various configurations provided by systemd-coredump. Below is a link to it.
 https://www.freedesktop.org/software/systemd/man/coredump.conf.html
User's might want  some bits of it be part of ConfigDB. 

The PR I was referring to is [PR#729](https://github.com/Azure/sonic-utilities/pull/729
) which enables kernel core dump feature. For this feature, it is desirable that users have an enable/disable knob as kdump requires dedicated 512MB of memory.

Reply:
Let me add Guohan to this thread. What we need is the "requirement/use case"? I don't see one.

I find it rather *risky* to have a DB variable to control, as it could get saved and persist across reboots, transparently, which is a *big* risk as it is likely to block core dumps unintentionally.

I would rather have this ability as a CLI tool, which disables temporarily and all should be back to default/enabled state upon reboot.

What we *do* need is the ability to limit count of cores per process and overall disk size taken by cores.